Homework_algs4/Homework_algs4/BinaryTree.cs

```diff
@@ -0,0 +1,304 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Homework_algs4
+{
+    class BinaryTree
+    {
+        public BinaryTreeNode RootNode { get; set; }
```

Comment:
Сеттер в коде надо сделать приватным.
```suggestion
        public BinaryTreeNode RootNode { get; private set; }
```